xscen/aggregate.py

```diff
@@ -123,43 +231,162 @@ def climatological_mean(
         and min_periods is None
     ):
         min_periods = window - 1
+
+    # unpack min_periods as fraction of window
+    if isinstance(min_periods, float):
+        if 0 < min_periods <= 1:
+            min_periods = int(np.floor(min_periods * window))
+        else:
+            raise ValueError(
+                f"When 'min_periods' is passed as a 'float', it must be between 0 and 1. Got {min_periods}."
+            )
+
+    # set min_periods
     min_periods = min_periods or window
     if min_periods > window:
         raise ValueError("'min_periods' should be smaller or equal to 'window'")
 
+    # if op is a dict, unpack it
+    if isinstance(op, dict):
+        op, op_kwargs = list(op.items())[0]
+        op_kwargs.setdefault("keep_attrs", True)
+    else:
+        op_kwargs = {
+            "keep_attrs": True
+        }  # ToDo: Is there a global setting in xscen so we don't need this?
```

Comment:
I think that calling `with xr.set_options(keep_attrs=True):` would work in a workflow. 

This is something we discussed a while ago, and having a blanket statement for `keep_attrs=True` would probably not be advised. It's better to do it as needed, such as here.

```suggestion
        }
```